### PR TITLE
DAOS-5990 test: Fixed container/open.py by adding pool.connect after …

### PR DESCRIPTION
**Pull request by shimizukko**
…container.create

TestContainer.create disconnects the pool, so need to reconnect because we're calling container.open

Also refactored.

Test-tag: container_open
Skip-unit-test: true
Skip-func-hw-test: true
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

**Review by amjustin13 — APPROVED** [review]
LGTM

File: src/tests/ftest/container/open.py [hunk 1]
```diff
@@ -121,20 +118,14 @@ def test_container_open(self):
...
-            self.pool[-1].connect()
-            self.container.append(
-                TestContainer(pool=self.pool[-1],
-                              daos_command=DaosCommand(self.bin)))
-            self.container[-1].get_params(self)
-            self.container[-1].create()
+            self.pool.append(self.get_pool())
+            self.container.append(self.get_container(pool=self.pool[-1]))
             container_uuids.append(uuid.UUID(self.container[-1].uuid))
+            self.pool[-1].connect()
```
Comment (phender) [inline src/tests/ftest/container/open.py:128]: Have you considered adding a `self.pool.connect()` to the [`TestContainer.open()`](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/test_utils_container.py#L449) method? It seems like that would be a more general solution than fixing this one test.

File: src/tests/ftest/container/open.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 hosts:
   test_servers:
     - server-A
+timeout: 340
```
Comment (phender) [inline src/tests/ftest/container/open.yaml:4]: This timeout seems very large based upon these test results:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3803/2/artifact/Functional/job-2020-11-06T01.31-f355574-container-open/results.html
```
container-gooduuid-goodpoh-hosts-pool-server_config-7b37   PASS	   31.24
container-gooduuid-badpoh-hosts-pool-server_config-add7    PASS	   28.70
container-baduuid-goodpoh-hosts-pool-server_config-0a92    PASS	   28.74
container-baduuid-badpoh-hosts-pool-server_config-6848	   CANCEL  22.30
```

**Review by phender — CHANGES_REQUESTED** [review]
I realize this is probably out of the scope of the original PR, but the idea of just calling `self.cancel` for the `container-baduuid-badpoh-hosts-pool-server_config` test case is a bit unsettling.  If we truly don't want to run a test then we shouldn't write it and using the `CANCEL` state may cause accounting inconsistencies with tests that are skipped for a ticket.  More importantly, even if the actions are odd or not recommended we should still verify the result and ensure it doesn't change from what is expected.

File: src/tests/ftest/container/open.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/container/open.py]: Done

File: src/tests/ftest/container/open.yaml (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/container/open.yaml]: I usually get the test time from the output near the end like below.
Total test time: 174s

Then I multiply it by 1.5 and use that value for timeout.

**Review by amjustin13 — APPROVED** [review]
LGTM

File: src/tests/ftest/container/open.yaml (hunk 2 above)
Comment (phender) [reply src/tests/ftest/container/open.yaml]: Yes, but the timeout is per test variant.  Using your '1.5' rule the timeout should be set using `max(31.24, 28.70, 28.74, 22,30) * 1.5 = 48`

**Review by phender — CHANGES_REQUESTED** [review]
Timeout needs to be reduced to account for the run time of each test variant, not their combined runtime.

**Review by phender — APPROVED** [review]
Approving the changes with the agreement that the timeout will be fixed in the next PR.

File: src/tests/ftest/container/open.yaml (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/container/open.yaml]: Thanks. It looks like these tests are executed with fairly consistent amount of time, so fixing the timeout value isn't urgent. I ran the whole set of tests in CI, so I'd rather fix this in the next PR.

**Comment by shimizukko** [comment]
Failed Test:
Functional_Hardware_Large / 1-./osa/osa_offline_parallel_test.py:OSAOfflineParallelTest.test_osa_offline_parallel_test;single-properties-single-hosts-pool-1KB-servers-server_manager-778d – OSAOfflineParallelTest -> DAOS-6054

**Pull request merged**